library/init/data/int/bitwise.lean

```diff
@@ -0,0 +1,63 @@
+/-
+Copyright (c) 2017 Microsoft Corporation. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Author: Mario Carneiro
+-/
+
+prelude
+import init.data.int.basic init.data.nat.bitwise
+
+universe u
+
+namespace int
+
+  def test_bit : ℤ → ℕ → bool
+  | (m : ℕ) n := nat.test_bit m n
+  | -[1+ m] n := bnot (nat.test_bit m n)
+
+  def nat_bitwise (f : bool → bool → bool) (m n : ℕ) : ℤ :=
```

Comment:
I'm torn on whether the definitional equality should be the reduction into positive/negative cases as I've done here, or the theorems `bitwise or = bor` etc. Either way we will have both, but definitional reduction can be important here.